docs/upgrading-to-8.x.asciidoc

```diff
@@ -0,0 +1,233 @@
+[[upgrading-to-8.x]]
+=== Upgrade to version {version}
+
+Version {version} of Elastic APM offers a number of benefits over previous versions:
+
+* Reduced number of fields per index, better space efficiency, and faster queries
+* More granular data control
+* Errors and metrics data streams are shared with other data sources -- which means better long-term integration with the logs and metrics apps
+* Removes template inheritance for index lifecycle policies and makes use of new {es} index and component templates
```

Comment:
I'm not really sure what to put here. I stuck with the data stream benefits as those do fit.

Reply:
We don't have to motivate the changes from legacy to v2 index&component templates here, as this is a change that ES requires. I suggest to keep it brief and link to the ES docs. 
In general, component templates allow to build reusable template blocks that can then be used in multiple index templates. The definition order defines which component overwrites other component's definition for the same settings. It gives a lot of flexibility and ES also offers a [simulation API](https://www.elastic.co/guide/en/elasticsearch/reference/current/simulate-multi-component-templates.html) to test the resulting index template.